// @cloudbase/js-sdk/utilities/dist/esm/adapters/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as Web from './platforms/web';
import { isArray } from '../libs/util';
export var RUNTIME;
(function (RUNTIME) {
    RUNTIME["WEB"] = "web";
    RUNTIME["WX_MP"] = "wx_mp";
})(RUNTIME || (RUNTIME = {}));
export function useAdapters(adapters, options) {
    var adapterList = isArray(adapters)
        ? adapters
        : [adapters];
    for (var _i = 0, adapterList_1 = adapterList; _i < adapterList_1.length; _i++) {
        var adapter = adapterList_1[_i];
        var isMatch = adapter.isMatch, genAdapter = adapter.genAdapter, runtime = adapter.runtime;
        if (isMatch()) {
            return {
                adapter: __assign({ isMatch: isMatch }, genAdapter(options)),
                runtime: runtime,
            };
        }
    }
}
export function useDefaultAdapter() {
    return {
        adapter: __assign({}, Web.genAdapter()),
        runtime: RUNTIME.WEB,
    };
}
